extensions/elasticsearch-rest-client/runtime/src/main/java/io/quarkus/elasticsearch/restclient/lowlevel/runtime/ElasticsearchConfig.java

```diff
@@ -16,7 +17,7 @@
      * The list of hosts of the Elasticsearch servers.
      */
     @ConfigItem(defaultValue = "localhost:9200")
-    public List<String> hosts;
+    public List<InetSocketAddress> hosts;
```

Comment:
From the javadoc:

```
 * This class implements an IP Socket Address (IP address + port number)
 * It can also be a pair (hostname + port number), in which case an attempt
 * will be made to resolve the hostname.
```

Do you really want that?

Reply:
We use `InetSocketAddress` in Quarkus for all the host:port combinations. I think, if I use `getHostString()` as I did, it does not resolve anything. At least that's what the Javadoc says.

Reply:
I checked (statically) what `java.net.InetSocketAddress#InetSocketAddress(java.lang.String, int)` calls and ended up in `java.net.InetAddress#getAddressesFromNameService`...  Might be worth a double-check.